src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.netcoreapp.cs

```diff
@@ -64,126 +63,83 @@ public void TestNonAzureBlockingPeriod(string description, object[] Params)
             {
-                using (SqlConnection sqlConnection = new SqlConnection(connStr))
-                {
-                    try
-                    {
-                        sqlConnection.Open();
-                        throw new Exception("Connection Open must expect an exception!");
-                    }
-                    catch (SqlException e)
-                    {
-                        // if it is the first time the exception is happening (previousConnectionId == Guid.Empty) skip the check
-                        if (previousConnectionId != Guid.Empty)
-                        {
-                            switch (policy)
-                            {
-                                case PoolBlockingPeriod.Auto:
-                                case PoolBlockingPeriod.AlwaysBlock:
-                                    if (e.ClientConnectionId != previousConnectionId)
-                                    {
-                                        throw new Exception($"Connection Open with Policy '{policy}' expect an exception with same connection id!");
-                                    }
-                                    break;
-                                case PoolBlockingPeriod.NeverBlock:
-                                    if (e.ClientConnectionId == previousConnectionId)
-                                    {
-                                        throw new Exception($"Connection Open with Policy '{policy}' expect an exception with different connection id!");
-                                    }
-                                    break;
-                            }
-                        }
-                        previousConnectionId = e.ClientConnectionId;
-                    }
-                }
-                count++;
+                case PoolBlockingPeriod.Auto:
+                case PoolBlockingPeriod.AlwaysBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 0, firstErrorTimeInSecs);
+                    break;
+                case PoolBlockingPeriod.NeverBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 1, int.MaxValue);
+                    break;
             }
         }
 
         public void PoolBlockingPeriodAzureTest(string connStr, PoolBlockingPeriod? policy)
         {
             SqlConnection.ClearAllPools();
-            Guid previousConnectionId = Guid.Empty;
-            int count = 0;
+            int firstErrorTimeInSecs = GetConnectionOpenTimeInSeconds(connStr);
+            int secondErrorTimeInSecs = GetConnectionOpenTimeInSeconds(connStr);
+            switch (policy)
+            {
+                case PoolBlockingPeriod.AlwaysBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 0, firstErrorTimeInSecs);
+                    break;
+                case PoolBlockingPeriod.Auto:
+                case PoolBlockingPeriod.NeverBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 1, int.MaxValue);
```

Comment:
Instead of int.MaxValue, you could check for 2*Connection Timeout. If you are not setting the connection timeout, you could use a default connection timeout of 15 sec.